models/daily_rates.sql

```diff
@@ -77,13 +85,41 @@ rates as (
             rate_sheet_daily.effective_rate,
             lag(rate_sheet_daily.effective_rate) ignore nulls over (partition by base.usage_type order by base.date),
             lead(rate_sheet_daily.effective_rate) ignore nulls over (partition by base.usage_type order by base.date)
-        ) as effective_rate
+        ) as effective_rate,
+        base.usage_type like 'overage-%' as is_overage_rate,
+        replace(base.usage_type, 'overage-', '') as associated_usage_type,
+        case
+            when remaining_balance_daily.is_account_in_overage and is_overage_rate then 1
+            when not remaining_balance_daily.is_account_in_overage and not is_overage_rate then 1
+            else 0
+        end as rate_priority
+
     from base
+    inner join remaining_balance_daily
+        on base.date = remaining_balance_daily.date
     left join rate_sheet_daily
         on base.date = rate_sheet_daily.date
             and base.usage_type = rate_sheet_daily.usage_type
+),
+
+rates as (
+    select
+        date,
+        usage_type,
+        associated_usage_type,
+        service_type,
+        effective_rate,
+        is_overage_rate
+    from rates_w_overage
+    qualify row_number() over (partition by date, service_type, associated_usage_type order by rate_priority desc) = 1
 )
 
-select *
+select
+    date,
+    associated_usage_type as usage_type,
+    service_type,
+    effective_rate,
+    is_overage_rate,
+    row_number() over (partition by service_type, associated_usage_type order by date desc) = 1 as is_latest_rate
 from rates
 order by date
```

Comment:
@john-jerome if you want to validate this it's ready:

```sql
select *
from daily_rates
where usage_type='compute'
order by date desc
```

there's now a single compute rate per day, no more having one record for `overage-compute` and one for `compute`:

<img width="1157" alt="CleanShot 2023-01-06 at 13 45 40@2x" src="https://user-images.githubusercontent.com/11096727/211096983-5c11339a-ddfc-4e29-ab5b-81f204c0e4d4.png">

If you want to reconcile it against your bill, you can do:

```sql
```sql
-- model
select
    date,
    service,
    sum(spend) as spend
from daily_spend
where service='Compute'
group by 1,2
order by date desc, spend desc
;

-- actual
select
    usage_date as date,
    usage_type as service,
    usage_in_currency as spend
from snowflake.organization_usage.usage_in_currency_daily
where 
    account_locator='<your_account_locator>' -- Not required if you only have 1 account
    and usage_type='compute'
order by date desc, spend desc
;
```

I found 1 day with a discrepancy when doing this check. On the day the account entered overage, there are two line items in `usage_in_currency_daily`. One for `overage` and one for `overage-compute`. This is a bug in our model, which assumes the account was in overage for the entire day because it ended the day with 0 balance. This will be very tricky to fix, but i will open an issue for it regardless. Outside the scope of this PR.

Reply:
@john-jerome i'm going to merge this since and cut a new release, but will keep an eye on this thread and implement any additional fixes required if you still notice issues

Reply:
hey @ian-whitestone, only saw this now (I'm based in Europe 😄 )

Both `daily_rates` and `daily_spend` models look good now, as you've mentioned there's now only one compute rate per day, and totals match (I checked on the monthly level)
Thanks again for addressing this!

Reply:
Thanks @john-jerome for checking! Much appreciated. Enjoy the weekend!